pyiron_contrib/atomistics/atomistics/job/trainingcontainer.py

```diff
@@ -145,13 +176,18 @@ def include_dataset(self, dataset):
             - name: human readable name of the structure
             - atoms(:class:`ase.Atoms`): the atomic structure
             - energy(float): energy of the whole structure
+            - totalcharge(float): charge of the whole structure
             - forces (Nx3 array of float): per atom forces, where N is the number of atoms in the structure
+            - charges (Nx3 array of floats):
             - stress (6 array of float): per structure stress in voigt notation
         """
         self._table_cache = self._table.append(dataset, ignore_index=True)
         # in case given dataset has more columns than the necessary ones, swallow/ignore them in *_
-        for name, atoms, energy, forces, stress, *_ in dataset.itertuples(index=False):
-            self._container.add_structure(atoms, name, energy=energy, forces=forces, stress=stress)
+        for name, atoms, energy, totalcharge, forces, charges, stress, *_ in dataset.itertuples(index=False):
```

Comment:
Since we're now talking about columns that may or may not be defined in the dataframe passed in, it's probably better to switch over to `iterrows` and then check which columns are defined/not None.